lib/Doctrine/ORM/Query/Parser.php

```diff
@@ -949,29 +954,43 @@ public function AliasIdentificationVariable()
     }
 
     /**
-     * AbstractSchemaName ::= identifier
+     * AbstractSchemaName ::= fully_qualified_name | aliased_name | identifier
      *
      * @return string
      */
     public function AbstractSchemaName()
     {
-        $this->match(Lexer::T_IDENTIFIER);
-
-        $schemaName = ltrim($this->lexer->token['value'], '\\');
-
-        if (strrpos($schemaName, ':') !== false) {
-            list($namespaceAlias, $simpleClassName) = explode(':', $schemaName);
-
-            $schemaName = $this->em->getConfiguration()->getEntityNamespace($namespaceAlias) . '\\' . $simpleClassName;
+        if ($this->lexer->isNextToken(Lexer::T_FULLY_QUALIFIED_NAME)) {
```

Comment:
Sorry if my review sounds like bashing your efforts, but here we're breaking BC.
Currenlty we support users to do: `AppBundle:Namespaced\MyEntity` which would now not be supported anymore (because the regex rule would reject it).